pkg/apis/core/validation/validation_test.go

```diff
@@ -5273,6 +5273,172 @@ func TestPVCVolumeMode(t *testing.T) {
 	}
 }
 
+func TestValidateReadOnlyPersistentDisks_Enabled_Only_Disk(t *testing.T) {
+	// gate enabled with read-only disk, nil old => OK
+	fldPath := field.NewPath("testField")
+	utilfeature.DefaultMutableFeatureGate.Set(fmt.Sprintf("%v=true", features.SkipReadOnlyValidationGCE))
+	defer utilfeature.DefaultMutableFeatureGate.Set(fmt.Sprintf("%v=false", features.SkipReadOnlyValidationGCE))
+	volumes := []core.Volume{
+		{
+			GCEPersistentDisk: &core.GCEPersistentDiskVolumeSource{
+				ReadOnly: true,
+			},
+		},
+	}
+	errors := ValidateReadOnlyPersistentDisks(volumes, nil, fldPath)
+	if len(errors) != 0 {
+		t.Errorf("unexpected errors: %v", len(errors))
+	}
+}
+
+func TestValidateReadOnlyPersistentDisks_Enabled_Write_Disk(t *testing.T) {
```

Comment:
Can you turn these into a table-test like most of the others?  The common pattern is like:

```
cases := []struct {
	name string
	gateValue bool
	input []core.Volume
	oldState []core.Volume
	expectError bool
}{{
	name: "gate off, create, read-only",
	input: []core.Volume{{
		GCEPersistentDisk: &core.GCEPersistentDiskVolumeSource{
			ReadOnly: true,
		},
	}},
	expectError: false,
},{
	name: "gate on, create, read-only",
	gateValue: true,
	input: []core.Volume{{
		GCEPersistentDisk: &core.GCEPersistentDiskVolumeSource{
			ReadOnly: true,
		},
	}},
	expectError: false,
},{
	name: "gate off, create, read-write",
	input: []core.Volume{{
		GCEPersistentDisk: &core.GCEPersistentDiskVolumeSource{
			ReadOnly: false,
		},
	}},
	expectError: true,
},{
	name: "gate on, create, read-write",
	gateValue: true,
	input: []core.Volume{{
		GCEPersistentDisk: &core.GCEPersistentDiskVolumeSource{
			ReadOnly: false,
		},
	}},
	expectError: false,
}, { 
```

and so on.  Then below that, just loop over each case with `t.Run()`, set the
gate value, call the function, check for error or non-errors based on the case.

Reply:
Okay, I'll finish it